frontend/src/sdk/actions.ts

```diff
@@ -114,17 +114,35 @@ export const approveCreateDepositStake = async (geyserAddress: string, amount: B
 
   const salt = randomBytes(32)
   const vaultAddress = await router.callStatic.create2Vault(config.VaultFactory.address, salt)
-  const vault = new Contract(vaultAddress, config.VaultTemplate.address, signer)
-
-  const lockPermission = await signPermission('Lock', vault, signer, geyserAddress, token.address, amount, 0)
-
+  const vault = new Contract(vaultAddress, config.VaultTemplate.abi, signer)
+  const lockPermission = await signPermission('Lock', vault, signer, geyserAddress, token.address, amount, '0')
   const args = [geyserAddress, config.VaultFactory.address, await signer.getAddress(), amount, salt, lockPermission]
 
   await token.approve(router.address, amount)
 
   return router.create2VaultAndStake(...args) as Promise<TransactionResponse>
 }
 
+export const depositStake = async (
```

Comment:
if i understand correctly, this version of combining deposit and stake allows us to pass in the user's vault address (leaving the creation up to us) such as in `create()` as opposed to in `permitCreateDepositStake` where a new vault gets created for each transaction?

Reply:
Yes, and to be more precise, I renamed the function to `approveDepositStake`

In the context of the app, if the user does not have a vault, staking will go through `approveCreateDepositStake`, which will create the vault for them. For subsequent stakes, it will go through `approveDepositStake` since we will have a vault address.

`permitCreateDepositStake` and `permitDepositStake` are the counterparts for permit-enabled ERC20 tokens, but it seems they are not fully implemented yet

Reply:
The name change sounds perfect!